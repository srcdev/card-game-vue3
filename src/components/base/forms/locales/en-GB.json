{
  "counterBetweenMinAndMax": "{ count } of between { min } and { max }",
  "counterNofMax": "{ count } of { max }",
  "optionalText": "(Optional)",
  "requiredText": "(Required)"
}
