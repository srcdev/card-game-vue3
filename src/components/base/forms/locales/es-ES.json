{
  "counterBetweenMinAndMax": "ES: { count } of between { min } and { max }",
  "counterNofMax": "ES: { count } of { max }",
  "optionalText": "(ES: Optional)",
  "requiredText": "(ES: Required)"
}
