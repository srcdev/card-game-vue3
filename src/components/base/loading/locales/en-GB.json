{
  "loading.text": "EN: Loading &hellip;"
}
