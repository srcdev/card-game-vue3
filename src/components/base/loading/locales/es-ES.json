{
  "loading.text": "ES: Loading &hellip;"
}
