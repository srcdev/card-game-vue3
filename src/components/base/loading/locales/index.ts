import enGB from "./en-GB.json";
import esES from "./es-ES.json";

export const translations: any = {
  "en-GB": enGB,
  "es-ES": esES,
};
export default translations;
