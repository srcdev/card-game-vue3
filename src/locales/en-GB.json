{
  "websiteTitle": "Fill In The Blanks Card Game",
  "joinGameTitle": "Join Game",
  "joinGameExplainer": "Before you joiong the game, please enter a name for the others to know who's playing",
  "joinGameFormUserNameLabel": "Your name",
  "joinGameFormUserNamePlaceholder": "eg: Norbert",
  "joinGameFormUserNameHint": "Example { example }",
  "joinGameFormUserNameError": "",
  "joinGameFormSubmit": "Join game",
  "errors": {
    "username": "A valid username is required"
  }
}
