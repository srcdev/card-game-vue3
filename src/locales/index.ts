import enGB from "./en-GB.json";

export const translations: any = {
  "en-GB": enGB
};
export default translations;
