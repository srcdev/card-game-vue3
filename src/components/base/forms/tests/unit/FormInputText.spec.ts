import { mount } from "@vue/test-utils";
import { it, describe, expect } from "vitest";
import { translations } from "../../locales/index";
import Component from "../../FormInputText.vue";
import { createPinia } from "pinia";
import { createI18n } from "vue-i18n";

//describe("Mounted Form Input Text component", () => {
it("Test Form Input Text component", async () => {
  const pinia = createPinia();
  const locale: any = "en-GB";

  const i18n = createI18n({
    legacy: false,
    locale: locale,
    messages: translations,
  });

  const propsData = {
    dataTestid: "dataTestid",
    inputType: "text",
    inputTestid: "inputTestid",
    inputName: "inputName",
    inputLabel: "inputLabel",
    inputHint: "inputHint",
    // inputHasErrors: {
    //   type: Boolean,
    //   default: false,
    // },
    inputErrorPattern: "inputErrorPattern",
    inputPattern: "emailaddress",
    inputPlaceholder: "inputPlaceholder",
    // inputRequired: {
    //   type: Boolean,
    //   default: false,
    // },
    // inputHasCounter: {
    //   type: Boolean,
    //   default: false,
    // },
    // inputPreventAutoFill: {
    //   type: Boolean,
    //   default: false,
    // },
    // submitAttempted: {
    //   type: Boolean,
    //   default: false,
    // },
    // modelValue: {
    //   type: String,
    //   default: "",
    // },
    // value: {
    //   type: String,
    //   default: "",
    // },
    // inputDisabled: {
    //   type: Boolean,
    //   default: false,
    // },
  };
  const wrapper = mount(Component, {
    global: {
      plugins: [i18n, pinia],
    },
    props: propsData,
  });

  expect(wrapper.vm).toBeTruthy();

  // Test element label
  expect(wrapper.get(`[data-testid="${propsData.dataTestid}-${propsData.inputTestid}-Label"]`).text()).toEqual(
    propsData.inputLabel
  );
  expect(
    wrapper.get(`[data-testid="${propsData.dataTestid}-${propsData.inputTestid}-Label"]`).text()
  ).toMatchSnapshot();

  // Test error state
  await wrapper.setProps({
    inputHasErrors: true,
  });

  expect(wrapper.get(`[type="${propsData.inputType}"]`).classes()).toContain("error");
  expect(wrapper.get(`label`).classes()).toContain("error");
});
//});
