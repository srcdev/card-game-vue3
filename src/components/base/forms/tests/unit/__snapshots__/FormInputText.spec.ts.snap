// Vitest Snapshot v1

exports[`Test Form Input Text component 1`] = `"inputLabel"`;
